### ISSUE TITLE
Updates and Changes in Q4 Flow

### ISSUE THREAD

**Issue by nmcdavid** [USAID-OHA-SI/q4_or_bust#14]
Site level hyperfile
- Added Nat/subnat to the site level hyperfile (added to the first union)

Both PSNU and Site level hyperfile
- Net New Targets
    - created FY22 Targets field
    - created FY21 Cumulative field
    - created "2022" field by calculating FY22 NET NEW Targets (FY22 Targets - FY21 Cumulative)
    - Added aggregate step before pivot
- Updated exclusion filters for VLC to make sure FY21 Q4 was included for steps
    - Correct FY & Q step
    - Remove Fields
- added filter for Previous Cumulative step to exclude current fiscal year (2021) - need to be updated every fiscal year
- Updated filter in current cumulative to reference FY21 Q4
- Added known issues field based on known issues tracker (flow is available in the hyperfile folder)

**Comment by kkarimovahashkes** [comment]
@achafetz @baboyma @tessam30 The "PSNUxIM Quarterly MSD Hyperfile Flow (.hyper MSDs) - with Known issues field Part 1" flow that is shared in this [folder ](https://drive.google.com/drive/u/0/folders/1JdCxYBdrXa9Hb2kMt1nXKPBznQMxPHUO)is not the latest one as the "Net New Targets" step doesn't include changes mentioned above by Nashiva. Could you please urgently share the latest flow? I'm not able to test the output of the Python code for Q4 data without the latest Tableau Prep flows used by Nashiva to produce the Q4 hyperfile. Many thanks

**Comment by achafetz** [comment]
Nashiva is out on leave until Dec 13. Unfortunately, the rest of us only have access to what is in the folder. The final output (FY21_Q4_v1.1_Output_PSNU_2021.11.23.zip) is located in the folder two levels up from the one you referenced has final dataset. This file is what is being used to populate the Tableau dashboards and can be used to QC against.

**Comment by roney-akter12** [comment]
@achafetz @baboyma @nmcdavid Could you please confirm that these are the only Q4 datasets PSNU_IM_FY19-22, PSNU_IM_FY15_18 and NAT_SUBNAT_FY15-22  used by Nashiva in her hyper file. Kholida is seeing few differences as she is comparing with Nashiva's output. 
Thank you for taking your time  and  looking into this!

**Comment by kkarimovahashkes** [comment]
@achafetz @baboyma @nmcdavid To clarify further, I also need a confirmation that files we are using to process Q4 data are correct ones. I'm currently downloading them from S3 and will soon share a link.

**Comment by kkarimovahashkes** [comment]
[Link ](https://drive.google.com/drive/folders/1zKybUdpgsaZrY6VNcswr_NkMi8Qx3UgP)with Q4 data inputs

**Comment by achafetz** [comment]
Correct, those are the MSD initial inputs. The clean versions will be available Jan 7.

**Comment by achafetz** [comment]
Sorry, in looking more closely, the archived version of the MSD is incorrect.  Also of note, the original files are stored as .txt files, not csv. You may run into an issue if you try reading them in as csv due  to "errant" commas.

You have: 
`MER_Structured_Datasets_PSNU_IM_FY15_18_20210917_v2_1`
It should be:
`MER_Structured_Datasets_OU_IM_FY15-18_20211112_v1_1`

**Comment by kkarimovahashkes** [comment]
@achafetz Thank you for confirmation, that what I found out as well. We were using incorrect archived version assuming that it doesn't change. But it seems it does. Running the flow and the code with latest version significantly reduced number of differences between the outputs.
